lib/ipam/ipam_block.go

```diff
@@ -82,6 +82,9 @@ func (b *allocationBlock) autoAssign(
 		// Affinity check is enabled but the host does not match - error.
 		s := fmt.Sprintf("Block affinity (%s) does not match provided (%s)", *b.Affinity, host)
 		return nil, errors.New(s)
+	} else if b.Affinity == nil {
```

Comment:
It's possible we hit this when trying to assign from non-affine blocks when all other blocks are full.

Should we protect against this?